### PR TITLE
Convert JSStackTraceTest to Kotlin

### PR DESCRIPTION
**Pull request by aymane-missouri-nw**
## Summary:
This PR converts JSStackTraceTest.java to Kotlin as requested in https://github.com/facebook/react-native/issues/37708.

## Changelog:
[INTERNAL] [CHANGED] - Convert JSStackTraceTest to Kotlin

## Test Plan:
Run ./gradlew :packages:react-native:ReactAndroid:test.
All tests should pass.

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/util/JSStackTraceTest.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     Assert.assertEquals(
+         message,
+         "Error, stack:\n"
+             + "method_from_bundle@7:11\n"
+             + "method_from_ram_bundle@199.js:18:13\n"
+             + "method_from_ram_bundle_with_address@199.js:18:13\n"
+             + "method_from_segment@seg-1.js:9:18\n"
+             + "method_from_segment_with_address@seg-1.js:9:18\n"
+             + "method_from_ram_segment@seg-3_198.js:10:20\n"
+             + "method_from_ram_segment_with_address@seg-3_198.js:10:20\n");
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/util/JSStackTraceTest.kt:94]: Can we use a multiline string here instead:
https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html#use-multiline-strings

**Review by cortinico — CHANGES_REQUESTED** [review]

**Comment by cortinico** [comment]
`test_android` is failing with:
```
com.facebook.react.util.JSStackTraceTest > testSymbolication FAILED
    org.junit.ComparisonFailure at JSStackTraceTest.kt:85
```

Can you look into it @aymane-missouri-nw ?

**Comment by aymane-missouri-nw** [comment]
@cortinico can you look into this PR

**Review by cortinico — APPROVED** [review]
LGTM 👍

**Pull request closed without merging**